airflow/providers/amazon/aws/operators/bedrock.py

```diff
@@ -91,3 +98,155 @@ def execute(self, context: Context) -> dict[str, str | int]:
+
+        self.job_name = job_name
+        self.custom_model_name = custom_model_name
+        self.role_arn = role_arn
+        self.base_model_id = base_model_id
+        self.training_data_config = {"s3Uri": training_data_uri}
+        self.output_data_config = {"s3Uri": output_data_uri}
+        self.hyperparameters = hyperparameters
+        self.ensure_unique_job_name = ensure_unique_job_name
+        self.customization_job_kwargs = customization_job_kwargs or {}
+
+        self.valid_action_if_job_exists: set[str] = {"timestamp", "fail"}
+
+    def execute_complete(self, context: Context, event: dict[str, Any] | None = None) -> str:
+        event = validate_execute_complete_event(event)
+
+        if event["status"] != "success":
+            raise AirflowException(f"Error while running job: {event}")
+
+        self.log.info("Bedrock model customization job `%s` complete.", self.job_name)
+        return self.hook.get_job_arn(event["job_name"])
+
+    def execute(self, context: Context) -> dict:
+        response = {}
+        retry = True
+        while retry:
+            # If there is a name conflict and ensure_unique_job_name is True, append the current timestamp
+            # to the name and retry until there is no name conflict.
+            # - Break the loop when the API call returns success.
+            # - If the API returns an exception other than a name conflict, raise that exception.
+            # - If the API returns a name conflict and ensure_unique_job_name is false, raise that exception.
+            try:
+                # Ensure the loop is executed at least once, and not repeat unless explicitly set to do so.
+                retry = False
+                self.log.info("Creating Bedrock model customization job '%s'.", self.job_name)
+
+                response = self.hook.conn.create_model_customization_job(
+                    jobName=self.job_name,
+                    customModelName=self.custom_model_name,
+                    roleArn=self.role_arn,
+                    baseModelIdentifier=self.base_model_id,
+                    trainingDataConfig=self.training_data_config,
+                    outputDataConfig=self.output_data_config,
+                    hyperParameters=self.hyperparameters,
+                    **self.customization_job_kwargs,
+                )
+            except ClientError as error:
+                if error.response["Error"]["Message"] != "The provided job name is currently in use.":
+                    raise error
+                if not self.ensure_unique_job_name:
+                    raise error
+                retry = True
+                self.job_name = f"{self.job_name}-{int(utcnow().timestamp())}"
+                self.log.info("Changed job name to '%s' to avoid collision.", self.job_name)
+
+        if response["ResponseMetadata"]["HTTPStatusCode"] != 201:
+            raise AirflowException(f"Bedrock model customization job creation failed: {response}")
```

Comment:
I've just wondering is it always return [201](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201) status code? And is this check required if we other errors didn't raised?

Reply:
I am only catching "job already exists" errors above so I can retry them, so i think this is still needed for other error cases:
![image](https://github.com/apache/airflow/assets/1920178/d780ccaf-9951-4ca5-9d99-393595bad16d)

Reply:
and yes, according to [their docs](https://docs.aws.amazon.com/bedrock/latest/APIReference/API_CreateModelCustomizationJob.html), a successful request will always return a 201:
![image](https://github.com/apache/airflow/assets/1920178/7b4329b6-30b9-42ab-9972-328340870ae2)